common/src/main/java/com/wynntils/features/inventory/InventoryEmeraldCountFeature.java

```diff
@@ -94,16 +90,16 @@ public void onContainerRender(ContainerRenderEvent event) {
             switch (emeraldCountType.get()) {
                 case TEXT -> renderTextCount(event.getPoseStack(), textX, y, topEmeralds);
                 case TEXTURE -> {
-                    int adjustedY;
-                    if (displayBottom) {
-                        int displayedTextures = getFilteredEmeraldAmounts(topEmeralds).length;
-                        int sizeY = displayedTextures * TEXTURE_SIZE + 2;
-                        int bottomY = containerScreen.imageHeight - TEXTURE_SIZE * 3 - 2;
-                        adjustedY = y - Math.max(sizeY - bottomY, 0);
-                    } else {
-                        adjustedY = y;
+                    if (displayBottom) { // ensure we don't overlap with bottom textures
+                        int topDisplayedTextureCount = Arrays.stream(getRenderableEmeraldAmounts(topEmeralds))
+                                .filter(s -> showZerosInEmeraldCount.get() || !s.equals("0"))
+                                .toArray()
+                                .length;
```

Comment:
nit: i believe there is a count method on stream; looks good otherwise

Reply:
I don't think it matters that much, count returns a long anyway so you'll have to cast

Reply:
mapToLong() then sum()

Reply:
Actually we can just mapToInt then sum. I will do that.

Reply:
Why not use the `.count()` and cast? This sum business just look funky.

Reply:
It is the same internally, just uses mapToLong instead

Reply:
What do you mean? When I check the JDK source code, the implementation for `count` I find is `evaluate(ReduceOps.makeRefCounting())`, which in turn seems to be highly optimized.

Reply:
OH I see count says *equivalent* to mapToLong